### PR TITLE
add lint and manpage check to make validate

### PR DESCRIPTION
**Pull request by baude**
make validate now runs golangci-lint and the man-page-checker to ensure
a PR is ready for our CI.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
requires -> https://github.com/openshift/release/pull/4968

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@edsantiago @mheon @giuseppe @vrothberg @TomSweeneyRedHat @QiWang19 @jwhonce PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tetss

File: .cirrus.yml [hunk 1]
```diff
@@ -104,7 +104,6 @@ gating_task:
         # N/B: entrypoint.sh resets $GOSRC (same as make clean)
         - '/usr/local/bin/entrypoint.sh install.tools |& ${TIMESTAMP}'
         - '/usr/local/bin/entrypoint.sh validate |& ${TIMESTAMP}'
-        - '/usr/local/bin/entrypoint.sh golangci-lint |& ${TIMESTAMP}'
```
Comment (edsantiago) [inline .cirrus.yml]: Would it make sense then to also remove `hack/man-page-checker` from line 113 (below, under `build_script`) ?

Comment (baude) [reply .cirrus.yml]: sure!

**Comment by edsantiago** [comment]
Um one more question: is `make validate` something that a regular dev would ever run in their environment? If so, are devs required to have `golangci-lint` installed? Because now it's a hard requirement.

**Comment by baude** [comment]
yeah it is basically now a requirement that you lint prior to submission unless of course you write perfect code?  that doesnt apply to me

**Pull request merged**